curriculum/challenges/english/25-front-end-development/lab-email-masker/66b205e6eacba4c4e54ea434.md

```diff
@@ -0,0 +1,68 @@
+---
+id: 66b205e6eacba4c4e54ea434
+title: Build an Email Masker
+challengeType: 14
+dashedName: build-an-email-masker
+---
+
+# --description--
+
+In this lab, you will mask the username part of an email address with asterisks. Masking is a term used to hide or replace sensitive information with asterisks or other characters.
+
+For example, if the email address was `myEmail@email.com`, then the masked email address will be `m*******l@email.com`.
+
+**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab. 
+
+**User Stories:**
+
+1. Declare a variable named `email` to store the email address you want to mask. 
+2. You should create a variable called `maskedName` and assign it the result of masking the `email`.  Remember that you can use methods like `slice`, `repeat`, `indexOf` `slice` or even `replace` to help you.
+3. Log the `maskedName` variable to the console. 
+
+# --hints--
+
+You should have an `email` variable.
+
+```js
+assert.isDefined(email);
+```
+
+You should assign `apple.pie@example.com` to your `email` variable.
+
+```js
+assert.equal(email, "apple.pie@example.com");
+```
+
+Your masked email address should look like this: `a*******e@example.com`.
```

Comment:
we should add another user story to make sure they don't hardcode the result like Tom mentioned earlier.
I would suggest experimenting with `assert.notMatch`